tools/downloader/main.py

```
@@ -0,0 +1,160 @@
+def main() -> int:
+    """メイン関数
+
+    Returns:
+        int: 終了コード（成功時は0、失敗時は1）
+    """
+    # 引数を解析
+    args = parse_arguments()
+
+    # validate arguments
+    if args.delay < MIN_DELAY:
+        parser.error(f"待機時間は {MIN_DELAY} 秒以上である必要があります")
```

Ruff (0.8.2)

Undefined name `parser`

(F821)